### PR TITLE
display a timeout modal

### PR DESCRIPTION
**Pull request by ryanwoldatwork**
this pops after 25 minutes so far

### CODE REVIEW

File: app/views/components/_timeout_modal.html.erb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  >
+    <div class="usa-modal__content">
+      <div class="usa-modal__main">
+        <h2 class="usa-modal__heading" id="modal-1-heading">
+          Are you still using Touchpoints?
+        </h2>
+        <div class="usa-prose">
+          <p id="modal-1-description">
+            For security reasons,
+            your session will timeout in 5 minutes due to inactivity.
```
Comment (sanason) [inline app/views/components/_timeout_modal.html.erb:25]: The timeout here isn't strictly an inactivity timeout, is it? It resets when you reload the page but not for ajax calls. So maybe it shouldn't say "due to inactivity".

File: app/views/components/_timeout_modal.html.erb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            Do you want to extend your session?
+          </p>
+           <div class="timer" id="timer">5:00</div>
+        </div>
+        <div class="usa-modal__footer">
+          <ul class="usa-button-group">
+            <li class="usa-button-group__item">
+              <a
+                href="<%= admin_forms_path %>"
```
Comment (sanason) [inline app/views/components/_timeout_modal.html.erb:34]: When I press this button, it closes the modal but it also autoscrolls to the bottom of the page. Any idea why? It's weird behavior.

**Review by sanason — COMMENTED** [review]
This is a nice improvement! Just a few comments.

File: app/views/components/_timeout_modal.html.erb (hunk 1 above)
Comment (ryanwoldatwork) [reply app/views/components/_timeout_modal.html.erb:25]: correct.

on Data Collections, there are measures to maintain a heartbeat and activity for ajax requests.

This PR does not address ajax, nor extend that pattern.

File: app/views/components/_timeout_modal.html.erb (hunk 2 above)
Comment (ryanwoldatwork) [reply app/views/components/_timeout_modal.html.erb:34]: good catch. will check it out

File: app/views/components/_timeout_modal.html.erb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        location.reload()
+      }
+    });
+  }
+
+  $(function() {
+    $("#extend-session").on("click", function(e) {
+      e.preventDefault();
+      $.get("/admin/forms");
+      timerOff = true;
```
[the remark this answers is not in the document]
Comment (ryanwoldatwork) [reply app/views/components/_timeout_modal.html.erb:103]: yep

[the remark this answers is not in the document]
Comment (ryanwoldatwork) [reply app/views/components/_timeout_modal.html.erb:103]: incorporated

**Pull request merged**